### ISSUE TITLE
iOS app doesn't crash when throw exception in CoroutineScope with Dispatchers.Default

### ISSUE THREAD

**Issue by belousovgm** [Kotlin/kotlinx.coroutines#2723]
Hi!
Why code like this:
```
CoroutineScope(Dispatchers.Default).launch {
throw IllegalStateException()
}
```
doesn't crash iOS app?

But 
```
CoroutineScope(Dispatcher.Main).launch {
    withContext(Dispatcher.Default) {
        throw IllegalStateException()
    }
}
```
crashes.

Kotlin: 1.4.31
Coroutines: 1.4.2-native-mt

**Comment by Thomas-Vos** [comment]
See the following: https://github.com/Kotlin/kotlinx.coroutines/issues/2398

**Comment by qwwdfsad** [comment]
Fixed in `1.5.1-native-mt`. For an additional explanation of the behaviour see https://github.com/Kotlin/kotlinx.coroutines/pull/2804#discussion_r667930024

**Comment by YoussefHachicha** [comment]
if fixed in 1.5.1-native-mt should it not happen in 1.8.0? because the issue is happening to me in 1.8.0 ??
